includes/Core/Authentication/Authentication.php

```diff
@@ -443,6 +451,47 @@ public function get_disconnect_url() {
 		);
 	}
 
+	/**
+	 * Gets the proxy permission url
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string Proxy permission url.
+	 */
+	private function get_proxy_permissions_url() {
+		return add_query_arg(
+			array(
+				'action' => Google_Proxy::ACTION_PERMISSIONS,
+				'nonce'  => wp_create_nonce( Google_Proxy::ACTION_PERMISSIONS ),
+			),
+			admin_url( 'index.php' )
+		);
+	}
+
+	/**
+	 * Handles proxy permissions
+	 *
+	 * @since n.e.x.t
+	 */
+	private function handle_proxy_permissions() {
+		if ( ! current_user_can( Permissions::AUTHENTICATE ) ) {
+			wp_die( esc_html__( 'You need a higher level of permission.', 'google-site-kit' ) );
+		}
+
+		if ( ! $this->credentials->using_proxy() ) {
+			wp_die( esc_html__( 'You need a higher level of permission.', 'google-site-kit' ) );
+		}
+
+		$nonce = $this->context->input()->filter( INPUT_GET, 'nonce' );
+		if ( ! wp_verify_nonce( $nonce, Google_Proxy::ACTION_PERMISSIONS ) ) {
+			wp_die( esc_html__( 'An error has occurred. Please reload the page and try again.', 'google-site-kit' ) );
+		}
+
+		wp_safe_redirect( $this->get_oauth_client()->get_proxy_permissions_url() );
+		exit;
+
+	}
```

Comment:
Rather than putting these two methods at an arbitrary point in the class, let's add them below where the related `handle_sync_site_fields` and `get_proxy_setup_url` methods are.